src/interceptors/ClientRequest/utils/createRequest.ts

```diff
@@ -20,6 +20,21 @@ export function createRequest(clientRequest: NodeClientRequest): Request {
     }
   }
 
+  /**
+   * Translate the authentication from the request URL to
+   * the request "Authorization" header.
+   * @see https://github.com/mswjs/interceptors/issues/438
+   */
+  if (clientRequest.url.username && clientRequest.url.password) {
```

Comment:
I've moved this logic to the `createRequest` function itself since it should always translate `ClientRequest` to a valid `Request` instance.